cloud_foundry_api/datadog_checks/cloud_foundry_api/cloud_foundry_api.py

```diff
@@ -0,0 +1,163 @@
+                return events
+
+            for cf_event in payload.get("resources", []):
+                try:
+                    dd_event, event_guid, event_ts = parse_event(cf_event, self._api_version)
+                except (ValueError, KeyError):
+                    self.log.exception("Could not parse event %s", cf_event)
+                    continue
+                # Stop going through events if we've reached one we've already fetched or if we went back in time enough
+                if event_guid == self._last_event_guid or int(time.time()) - event_ts > MAX_LOOKBACK_SECONDS:
+                    scroll = False
+                    break
+                # Store the event at which we want to stop on the next check run: the most recent of the current run
+                if event_ts > self._last_event_ts:
+                    self._last_event_guid = event_guid
+                    self._last_event_ts = event_ts
+                # Make sure we don't send duplicate events in case the pagination gets shifted by new events
+                if event_guid not in events:
+                    events[event_guid] = dd_event
+            # Fetch next page if any
+            next_url = get_next_url(payload, self._api_version)
+            if not next_url or not scroll:
+                break
+            page += 1
+        self.service_check(API_SERVICE_CHECK_NAME, CloudFoundryApiCheck.OK, tags=sc_tags)
+        return events
+
+    def get_events(self):
+        # type: () -> Dict[str, Event]
+        self.get_oauth_token()
+        if self._api_version == "v2":
+            params = {
+                "q": "type IN {}".format(self._event_filter),
+                "results-per-page": min(self._per_page, MAX_PAGE_SIZE_V2),
+                "order-direction": "desc",
+                "order-by": "timestamp",
+            }
+            headers = {"Authorization": "Bearer {}".format(self._oauth_token)}
+            url = "{}/v2/events".format(self._api_url)
+            return self.scroll_pages(url, params, headers)
+        elif self._api_version == "v3":
+            params = {
+                "types": self._event_filter,
+                "per_page": min(self._per_page, MAX_PAGE_SIZE_V3),
+                "order_by": "-created_at",
+            }
+            headers = {"Authorization": "Bearer {}".format(self._oauth_token)}
+            url = "{}/v3/audit_events".format(self._api_url)
+            return self.scroll_pages(url, params, headers)
+
+        self.log.error("Unknown api version `%s`, choose between `v2` and `v3`", self._api_version)
+        return {}
+
+    def check(self, _):
+        # type: (Dict[str, Any]) -> None
+        events = self.get_events()
+        tags = ["api_url:{}".format(self._api_url)] + self._tags
+        self.count("events.count", len(events), tags=tags)
+        for event in events.values():
+            self.event(event)
```

Comment:
Should `tags` be added to events being submitted as well?

Reply:
Oh yeah they probably should